Minio/ApiEndpoints/ObjectOperations.cs

```diff
@@ -690,8 +692,93 @@ private IObservable<Upload> listIncompleteUploads(string bucketName, string pref
                                                      objectName: objectName);
 
             var response = await this.ExecuteTaskAsync(this.NoErrorHandlers, request, cancellationToken);
+        }
+
+        /// <summary>
+        /// private helper method to remove list of objects from bucket
+        /// </summary>
+        /// <param name="bucketName"></param>
+        /// <param name="objectsList"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        private async Task<List<DeleteError>> removeObjectsAsync(string bucketName, List<DeleteObject> objectsList, CancellationToken cancellationToken)
+        {
+            string resource = "?delete";
+            var request = await this.CreateRequest(Method.POST, bucketName, resourcePath: resource);
+            List<XElement> objects = new List<XElement>();
+
+            for (int i = 0; i < objectsList.Count; i++)
+            {
+                objects.Add(new XElement("Object",
+                                       new XElement("Key", objectsList[i].Key)));
+            }
+
+            var deleteObjectsRequest = new XElement("DeleteObject", objects,
+                                        new XElement("Quiet",true));
```

Comment:
IObservable is closest to the channel returned in minio-go since Observable is a push based notification, that makes the RemoveObjectErrors available to the application as they are returned in an asynchronous fashion. If the method were synchronous, we could return an IEnumerable.

Reply:
@poornas I made this comment too soon. You are right. We still need to set quiet to true to receive errors during deletion of some of the objects in the list, right?

Reply:
@krisis , yes we are setting Quiet to true so that Delete errors can be processed